src/schema/users.ts

```diff
@@ -846,27 +846,42 @@ export const resolvers: IResolvers<any, Context> = {
         .getRawMany();
     },
     userStreak: async (_, __, ctx: Context, info): Promise<GQLUserStreak> => {
-      const streak = await graphorm.queryOneOrFail<GQLUserStreakTz>(
-        ctx,
-        info,
-        (builder) => ({
-          queryBuilder: builder.queryBuilder
-            .addSelect(
-              `(date_trunc('day', "${builder.alias}"."lastViewAt" at time zone COALESCE(u.timezone, 'utc'))::date) AS "lastViewAtTz"`,
-            )
-            .addSelect('u.timezone', 'timezone')
-            .innerJoin(User, 'u', `"${builder.alias}"."userId" = u.id`)
-            .where(`"${builder.alias}"."userId" = :id`, { id: ctx.userId }),
-          ...builder,
-        }),
-      );
+      try {
+        const streak = await graphorm.queryOneOrFail<GQLUserStreakTz>(
+          ctx,
+          info,
+          (builder) => ({
+            queryBuilder: builder.queryBuilder
+              .addSelect(
+                `(date_trunc('day', "${builder.alias}"."lastViewAt" at time zone COALESCE(u.timezone, 'utc'))::date) AS "lastViewAtTz"`,
+              )
+              .addSelect('u.timezone', 'timezone')
+              .innerJoin(User, 'u', `"${builder.alias}"."userId" = u.id`)
+              .where(`"${builder.alias}"."userId" = :id`, { id: ctx.userId }),
+            ...builder,
+          }),
+        );
 
-      const hasClearedStreak = await checkAndClearUserStreak(ctx, info, streak);
-      if (hasClearedStreak) {
-        return { ...streak, current: 0 };
-      }
+        const hasClearedStreak = await checkAndClearUserStreak(
+          ctx,
+          info,
+          streak,
+        );
+        if (hasClearedStreak) {
+          return { ...streak, current: 0 };
+        }
 
-      return streak;
+        return streak;
+      } catch (err) {
+        ctx.log.error('Error fetching user streak', err);
+        return {
+          max: 0,
+          total: 0,
+          current: 0,
+          lastViewAt: null,
+          userId: ctx.userId,
+        };
```

Comment:
I would not do it with catch because some legit error would then also return empty streak. I would rather early return if no streak in DB and return this empty stub.

Reply:
Removed the try...catch and just used same logic as inside the queryOneOrFail except without throwing the error.